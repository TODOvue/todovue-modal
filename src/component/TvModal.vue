<template>
  <div v-if="showModal" class="tv-modal-container" @click.self="animateModal">
    <div class="tv-modal" ref="tvModal">
      <div
        class="tv-modal-image tv-m-auto"
        :class="`tv-modal-${configModal.icon}`"
        v-if="configModal.icon"
      ></div>
      <div class="tv-modal-title" v-if="configModal.title">
        {{ configModal.title }}
      </div>
      <hr
        class="tv-modal-separator"
        v-if="configModal.title && configModal.description"
      />
      <div class="tv-modal-description" v-if="configModal.description">
        {{ configModal.description }}
      </div>
      <div class="tv-modal-actions">
        <tv-button
          isSuccess
          isSmall
          @click="acceptModal"
          v-if="configModal.confirmButtonText"
        >
          {{ configModal.confirmButtonText }}
        </tv-button>
        <tv-button
          isSmall
          @click="cancelModal"
          v-if="configModal.cancelButtonText"
        >
          {{ configModal.cancelButtonText }}
        </tv-button>
      </div>
    </div>
  </div>
</template>

<script>
import TvButton from "todovue-button";
import useModal from "@/composables/useModal";

export default {
  name: "TvModal",
  components: {
    TvButton,
  },
  props: {
    configModal: {
      type: Object,
      required: true,
    },
  },
  setup() {
    const {
      showModal,
      tvModal,
      animateModal,
      acceptModal,
      cancelModal,
      openModal,
    } = useModal();

    return {
      showModal,
      tvModal,
      animateModal,
      acceptModal,
      cancelModal,
      openModal,
    };
  },
  emits: ["acceptModal", "cancelModal"],
};
</script>

<style></style>
