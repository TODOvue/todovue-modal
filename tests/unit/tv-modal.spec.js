import { shallowMount } from "@vue/test-utils";
import TvModal from "@/component/TvModal.vue";
describe("useModal", () => {
  it("should matches snapshot", () => {
    const wrapper = shallowMount(TvModal, {
      props: {
        configModal: {},
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
