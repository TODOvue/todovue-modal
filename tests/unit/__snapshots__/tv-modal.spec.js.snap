// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useModal should matches snapshot 1`] = `<!--v-if-->`;
